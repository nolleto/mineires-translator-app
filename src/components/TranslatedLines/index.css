.translated-lines {
  margin: 10px 0;
}
