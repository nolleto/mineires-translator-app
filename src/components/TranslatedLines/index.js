import './index.css';

import PropTypes from 'prop-types';
import React from 'react';
import TranslatedLinesItem from '../TranslatedLinesItem';

const TranslatedLines = ({ lines }) => {
  const notEmpty = string => string.trim() !== '';
  const toTranslatedLinesItem = text => (
    <TranslatedLinesItem
      key={text}
      text={text}
    />
  );

  return (
    <div className='translated-lines'>
      {lines.filter(notEmpty).map(toTranslatedLinesItem)}
    </div>
  );
};

TranslatedLines.propTypes = {
  lines: PropTypes.arrayOf(PropTypes.string)
};

export default TranslatedLines;
