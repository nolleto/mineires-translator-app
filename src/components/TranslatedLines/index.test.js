import React from 'react';
import TranslatedLines from './index';
import { shallow } from 'enzyme';

const shallowComponent = (props = {}) => {
  const finalProps = {
    lines: [],
    ...props
  };

  return shallow(
    <TranslatedLines
      {...finalProps}
    />
  );
};

describe('TranslatedLines component', () => {
  describe('Snapshot', () => {
    it('when has no lines', () => {
      const snapshot = shallowComponent({ lines: [] });

      expect(snapshot).toMatchSnapshot();
    });

    it('when has lines', () => {
      const snapshot = shallowComponent({
        lines: [
          'line 1',
          'line 2',
        ]
      });

      expect(snapshot).toMatchSnapshot();
    });

    it('when has lines and empty lines', () => {
      const snapshot = shallowComponent({
        lines: [
          'line 1',
          'line 2',
          '',
          'lines 3'
        ]
      });

      expect(snapshot).toMatchSnapshot();
    });
  });
});
