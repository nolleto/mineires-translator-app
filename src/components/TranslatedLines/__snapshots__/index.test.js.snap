// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslatedLines component Snapshot when has lines 1`] = `
<div
  className="translated-lines"
>
  <TranslatedLinesItem
    key="line 1"
    text="line 1"
  />
  <TranslatedLinesItem
    key="line 2"
    text="line 2"
  />
</div>
`;

exports[`TranslatedLines component Snapshot when has lines and empty lines 1`] = `
<div
  className="translated-lines"
>
  <TranslatedLinesItem
    key="line 1"
    text="line 1"
  />
  <TranslatedLinesItem
    key="line 2"
    text="line 2"
  />
  <TranslatedLinesItem
    key="lines 3"
    text="lines 3"
  />
</div>
`;

exports[`TranslatedLines component Snapshot when has no lines 1`] = `
<div
  className="translated-lines"
/>
`;
